src/lambda_extract_metrics.py

```diff
@@ -38,14 +38,14 @@ def lambda_handler(event, context):
     logger.info(f"Event = {event}")
 
     # TODO: add a "semaphore" - if lambda works with a specific monitoring group, new invocation of lambda for the same monitoring group
-    # doesn't proceed    
+    # doesn't proceed
 
     settings_s3_path = os.environ["SETTINGS_S3_PATH"]
     iam_role_name = os.environ["IAMROLE_MONITORED_ACC_EXTRACT_METRICS"]
     timestream_metrics_db_name = os.environ["TIMESTREAM_METRICS_DB_NAME"]
     monitoring_group_name = event.get("monitoring_group")
 
-    settings = Settings.from_s3_path(settings_s3_path)
+    settings = Settings.from_s3_path(settings_s3_path, iam_role_name)
```

Comment:
I'd prefer to pass iam_role as named parameter.
Otherwise, it looks like IAM role is used to READ settings (while actually it is used to TRANSFORM they).

Reply:
Done